src/app/auth.tsx

```diff
@@ -0,0 +1,64 @@
+"use client";
+
+import React, { useEffect } from "react";
+import { usePathname, useRouter } from "next/navigation";
+import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";
+import { DatabaseAutoType } from "@/supabase";
+
+export const AuthRouting: React.FC<{ children: React.ReactNode }> = ({ children = <></> }) => {
+    const supabase = createClientComponentClient<DatabaseAutoType>();
+    const router = useRouter();
+    const pathname = usePathname();
+    const [loggedIn, setLoggedIn] = React.useState<boolean>();
+
+    // This fixes issues with navigation when the user has signed in but then signed out and caching still allows them to
+    // access pages. It also fixes authentication flow issues on sign in and sign out
+    const toRedirectTo = (): string | null => {
+        if (loggedIn === undefined) {
+            return null;
+        }
+
+        if (loggedIn) {
+            return window.location.pathname.startsWith("/login") ? "/clients" : null;
```

Comment:
I'd prefer 
```ts
if (loggedIn && ...("/login")) {
    return "/clients";
}
if (!loggedIn && ... "/login") {
    return "/login";
}
return null;
```
just reads a _lot_ better in my opinion

Reply:
very similar to how I wrote: https://github.com/VauxhallFoodbank/vauxhall-foodbank/blob/a6347c23f729be7903dd0192438f3214795a0dfa/src/middleware.ts#L13